awx/main/tasks.py

```diff
@@ -3137,3 +3009,212 @@ def deep_copy_model_obj(
             permission_check_func(creater, copy_mapping.values())
     if isinstance(new_obj, Inventory):
         update_inventory_computed_fields.delay(new_obj.id)
+
+
+class AWXReceptorJob:
+    def __init__(self, task, runner_params):
+        self.task = task
+        self.runner_params = runner_params
+
+        if not self.task.instance.is_container_group_task:
+            execution_environment_params = self.task.build_execution_environment_params(self.task.instance)
+            self.runner_params['settings'].update(execution_environment_params)
+
+    def run(self):
+        # Create a socketpair. Where the left side will be used for writing our payload
+        # (private data dir, kwargs). The right side will be passed to Receptor for
+        # reading.
+        sockin, sockout = socket.socketpair()
+
+        threading.Thread(target=self.transmit, args=[sockin]).start()
+
+        # We establish a connection to the Receptor socket and submit our work, passing
+        # in the right side of our socketpair for reading.
+        receptor_ctl = ReceptorControl('/var/run/receptor/receptor.sock')
+        result = receptor_ctl.submit_work(worktype=self.work_type,
+                                          payload=sockout.makefile('rb'),
+                                          params=self.receptor_params)
+        unit_id = result['unitid']
+
+        sockin.close()
+        sockout.close()
+
+        resultsock, resultfile = receptor_ctl.get_work_results(unit_id,
```

Comment:
this is expecting the return from here right?

https://github.com/project-receptor/receptor/blob/e293394235082df0b5f2cf037a43d13c6a1f3787/receptorctl/receptorctl/socket_interface.py#L233

I'd like to make it a little more clear what `resultfile` is, which comes from `self._sockfile = self._socket.makefile('rwb')`

There's 2 types of things being passed around, sockets and sockfiles. Some are made here, and some come from receptor. I'd like to try to get the naming to be as indicative of this as possible, like `result_sockfile`